src/metabase/models/setting.clj

```diff
@@ -527,9 +570,11 @@
                      database-local-value
                      env-var-value
                      db-or-cache-value
+                     (when-not *disable-init*
+                       init!)
                      default-value]]
```

Comment:
Since it's invalid to provide both an initialized value and a default, maybe it makes sense for readability (and therefore safety) to have the code reflect they are mutually exclusive here. For example:
```suggestion
                     (cond 
                       (:default setting) default-value
                       (:init setting) init!))]]
```

Reply:
Nice, this also makes the behavior safer until we add schema validation.

Reply:
Oops, I left out the `*disable-init*` from the suggestion but you still need it. Something like
```clojure
                     (cond 
                       (:default setting) default-value
                       (and (:init setting) (not *disable-init*)) init!))
```